kuma/javascript/src/bcd-signal.jsx

```diff
@@ -0,0 +1,809 @@
+        }
+
+        const signalApiUrl = '/api/v1/bc-signal';
+        const payload = {
+            slug: slug,
+            locale: locale,
+            feature: feature,
+            explanation: explanation,
+            browsers: browsers.join(', '),
+            supporting_material: supportingMaterial // eslint-disable-line camelcase
+        };
+
+        fetch(signalApiUrl, {
+            method: 'POST',
+            body: JSON.stringify(payload),
+            headers: {
+                'X-CSRFToken': getCookie('csrftoken'),
+                'Content-Type': 'application/json'
+            }
+        })
+            .then(function(response) {
+                if (!response.ok) {
+                    throw new Error('Something went wrong');
+                }
+                return response;
+            })
+            .then(function() {
+                bcSignalBlock.classList.remove('open');
+                bcSignalCompleteBlock.classList.add('open');
+            })
+            .catch(function() {
+                errorMessageWrapper.classList.add('visible');
+            });
+    };
+
+    /**
+     * Builds and creates round label for form control steps
+     * @param {Number} number - Number inside label, which is step index
+     * @param {Boolean} inline - Boolean defining if label should be inline
+     * @returns Step label as a `HTMLElement`
+     */
+    const createStepNumLabel = (number, inline) => {
+        const stepNumLabel = document.createElement('span');
+        let stepClass = 'step-num';
+        if (inline) {
+            stepClass += ' inline';
+        }
+
+        stepNumLabel.className = stepClass;
+        stepNumLabel.innerText = `${number}`;
+        return stepNumLabel;
+    };
+
+    /**
+     * Builds and wraps element in a form control from given element and parameters from controlObj
+     * @param {Object} controlObj - Object containing element that needs to be wrapped and related params
+     * @returns Form control as a `HTMLElement`
+     */
+    const createFormControl = controlObj => {
+        const control = document.createElement('div');
+        const controlInnerWrapper = document.createElement('div');
+        const controlHeader = document.createElement('div');
+        const controlDescription = document.createElement('div');
+
+        control.className = 'control';
+        controlInnerWrapper.className = 'control-wrap';
+        controlHeader.className = 'control-header';
+        if (controlObj.inline) {
+            controlHeader.className += ' has-control';
+        }
+        if (controlObj.optional) {
+            controlHeader.className += ' with-optional-label';
+        }
+
+        controlDescription.className = 'control-description';
+
+        controlHeader.appendChild(createStepNumLabel(controlObj.index, true));
+        controlHeader.appendChild(document.createTextNode(controlObj.header));
+        controlDescription.innerHTML = controlObj.description;
```

Comment:
Here's another concern with security. That `controlObj.description` is potentially coming from the l10n service which is maybe out of our control. So someone can fake the translation and inject whatever they want in the DOM for unsuspecting users.

Reply:
I see what you say, and makes sense to me. May I kindly ask, what would be your approach on solving this?